### PR TITLE
ASC-227 add molecularize

### PR DESCRIPTION
**Pull request by zreichert**
- molecularize script from rpc-openstack-system-tests
- convert to use click
- convert to use pytest
- add documentation

### CODE REVIEW

File: moleculerize/__init__.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import pkg_resources
+import click
+
+
+# ======================================================================================================================
+# Globals
+# ======================================================================================================================
+TEMPLATES_DIR = 'data'
+TEMPLATE = 'molecule.yml.j2'
+OUTPUT_FILE = 'molecule.yml'
```
Comment (ryan-rs) [inline moleculerize/__init__.py:19]: Look at those beautiful globals!

oh yeah

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Build molecule config files from an Ansible dynamic inventory file
+
+Quick Start Guide
+-----------------
+
+1. Install ``moleculerize`` from PyPI with pip::
+
+    $ pip install moleculerize
+
+2. For more information on using the Zigzag launch help by::
```
Comment (johnduarte) [inline README.rst]: s/the Zigzag/moleculerize/

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+python =
+3.6: py36
+3.5: py35
+2.7: py27
+
+[testenv:flake8]
+basepython = python
+skip_install = true
+deps = flake8
+commands = flake8 moleculerize.py tests
```
Comment (ryan-rs) [inline tox.ini]: Should be `flake8 moleculerize tests`.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+	rm -fr .tox/
+	rm -f .coverage
+	rm -fr htmlcov/
+	rm -fr .pytest_cache/
+
+clean-venv: check-venv ## remove all packages from current virtual environment
+	@source virtualenvwrapper.sh && wipeenv || echo "Skipping wipe of environment"
+
+lint: ## check style with flake8
+	flake8 moleculerize setup.py tests
```
Comment (ryan-rs) [inline Makefile:64]: You include `setup.py` in linting here, but not in the tox config.

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    $ moleculerize --help
+
+3. Here is an example of building a molecule config from an Ansible dynamic inventory file::
+
+    $ moleculerize /path/to_my/dynamic_inventory.json
```
Comment (johnduarte) [inline README.rst]: I think this should include the optional `--output` argument in order to make the result more explicit.

E.g.
```
    $ moleculerize --output molecule/default/molecule.yml /path/to/my/dynamic_inventory.json
```

File: README.rst [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+
+Release Process
+---------------
+
+See `release_process.rst`_ for information on the release process for 'zigzag'
```
Comment (johnduarte) [inline README.rst]: s/zigzag/moleculerize/

**Review by johnduarte — CHANGES_REQUESTED** [review]

**Review by johnduarte — APPROVED** [review]

File: README.rst (hunk 5 above)
Comment (zreichert) [reply README.rst]: fixed

File: README.rst (hunk 6 above)
Comment (zreichert) [reply README.rst]: fixed

File: README.rst (hunk 2 above)
Comment (zreichert) [reply README.rst]: fixed

File: moleculerize/__init__.py [hunk 7]
```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+# ======================================================================================================================
+# Imports
+# ======================================================================================================================
+import sys
+import json
+from argparse import ArgumentParser
```
Comment (ryan-rs) [inline moleculerize/__init__.py]: Unused import.

File: requirements.txt [hunk 8]
```diff
@@ -0,0 +1,11 @@
+-c constraints.txt
+mock
+pytest-rpc
```
Comment (ryan-rs) [inline requirements.txt]: This requirement is unnecessary.

File: moleculerize/__init__.py [hunk 9]
```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+# ======================================================================================================================
```
Comment (ryan-rs) [inline moleculerize/__init__.py]: Missing `__version__`.

File: setup.py [hunk 10]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env/python
```
Comment (ryan-rs) [inline setup.py]: Unsafe she-bang, use `#!/usr/bin/env python` instead.

File: moleculerize/__init__.py (hunk 9 above)
Comment (ryan-rs) [inline moleculerize/__init__.py]: Missing encoding declaration: `# -*- coding: utf-8 -*-`

File: MANIFEST.in [hunk 11]
```diff
@@ -0,0 +1,8 @@
+include CONTRIBUTING.rst
```
Comment (ryan-rs) [inline MANIFEST.in]: There is no `CONTRIBUTING.rst` included.

File: setup.py [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+from setuptools import setup
+
+with open('README.rst') as readme_file:
+    readme = readme_file.read()
+
+with open('HISTORY.rst') as history_file:
+    history = history_file.read()
+
+setup(
```
Comment (ryan-rs) [inline setup.py]: Missing requirements: `click` and `jinja2`. Can't run cmdline tool without these.

**Review by ryan-rs — CHANGES_REQUESTED** [review]
Can't run the cmdline tool, see comments.

File: moleculerize/__init__.py (hunk 1 above)
Comment (zreichert) [reply moleculerize/__init__.py:19]: I feel sick

File: moleculerize/__init__.py [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+def main(inv_file, template, output):
+    """Main program routine.
+
+    Args:
+        inv_file (str): The Ansible inventory file.
+        template (str): The template to use.
+        output (str): The target output file
+    Returns:
+        int: The exit code to return to the shell.
+    """
```
Comment (ryan-rs) [inline moleculerize/__init__.py]: When running `moleculerize --help` the docs are read from the doc string which results in unreadable output:

```
Usage: moleculerize [OPTIONS] INV_FILE

  Main program routine.

  Args:     inv_file (str): The Ansible inventory file.     template (str):
  The template to use.     output (str): The target output file Returns:
  int: The exit code to return to the shell.

Options:
  -t, --template TEXT  Molecule config template file
  -o, --output TEXT    Output file path for molecule config.
  --help               Show this message and exit.
```

See `zigzag.cli` for an example on how to format the output to make it pretty.

File: setup.py [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    packages=['moleculerize'],
+    install_requires=['Click>=6.0', 'jinja2'],
+    include_package_data=True,
+    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*',
+    license="Apache Software License 2.0",
+    zip_safe=False,
+    keywords='moleculerize',
+    classifiers=[
+        'Development Status :: 2 - Pre-Alpha',
+        'Framework :: Flake8',
```
Comment (ryan-rs) [inline setup.py]: Copy pasta!

**Review by ryan-rs — CHANGES_REQUESTED** [review]

**Review by ryan-rs — APPROVED** [review]

**Pull request merged**